### PR TITLE
Add new optimization rules to optimizations.md

### PR DESCRIPTION
**Pull request by dwightguth**
We measured which opcodes are most expensive. As a result of these measurements, we have identified 8 opcodes we want to try to add to optimizations.md. Four of them are included in this PR: ISZERO, JUMPDEST, JUMP, and JUMPI.

### CODE REVIEW

File: kevm-pyk/src/kevm_pyk/kproj/evm-semantics/optimizations.md [hunk 1]
```diff
@@ -315,6 +315,138 @@ module EVM-OPTIMIZATIONS
...
+            ...
+          </callState>
+          ...
+        </evm>
+        ...
+      </ethereum>
+      ...
+    </kevm>
+    requires ( Ghigh < SCHED > <=Gas GAVAIL )
+     andBool ( size( WS ) <=Int 1023 )
```
Comment (theo25) [inline kevm-pyk/src/kevm_pyk/kproj/evm-semantics/optimizations.md:446]: If I understand correctly, there is a mistake in this requires.

```suggestion
     andBool ( size( WS ) <=Int 1022 )
```

**Review by theo25 — APPROVED** [review]
Looks good but I left a suggested correction. Feel free to ignore if I have misunderstood.

File: kevm-pyk/src/kevm_pyk/kproj/evm-semantics/optimizations.md (hunk 1 above)
Comment (dwightguth) [reply kevm-pyk/src/kevm_pyk/kproj/evm-semantics/optimizations.md:446]: So, you might be right, but I'm actually pretty sure that these overflow checks are only required on opcodes that emit more stack values than they consume, of which this one is not one. Let me keep tinkering and I may end up making a modified version of this PR that just deletes the unnecessary checks.

**Review by theo25 — APPROVED** [review]
This looks good to me.

**Pull request merged**